src/main/java/nl/novi/techiteasy/controllers/TelevisionController.java

```diff
@@ -1,53 +1,79 @@
+    public ResponseEntity<Television> showTv(@PathVariable ("id") Long id) {
+        Optional<Television> television = televisionRepository.findById(id);
+        if (television.isEmpty()) {
+            throw new RecordNotFoundException("Id " + id + " not found ");
         } else {
-            televisionDatabase.set(id, tv);
-            return ResponseEntity.noContent().build();
+            Television foundTelevision = television.get();
+            return ResponseEntity.ok().body(foundTelevision);
         }
     }
 
-    @DeleteMapping("/deleteTv/{id}")
-    public ResponseEntity<Object> deleteTv(@PathVariable ("id") int id) {
-        televisionDatabase.set(id, null);
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Object> deleteTelevision (@PathVariable("id") Long id){
+        televisionRepository.deleteById(id);
         return ResponseEntity.noContent().build();
     }
-}
+
+    @PutMapping("/{id}")
+    public ResponseEntity<Television> updateTelevision (@PathVariable("id") Long id, @RequestBody Television
+    updatedTelevision) {
+        Optional<Television> television = televisionRepository.findById(id);
+            if (television.isEmpty()) {
+                throw new RecordNotFoundException("Id " + id + " not found ");
+            } else {
+                Television television1 = television.get();
+                television1.setAmbiLight(updatedTelevision.getAmbiLight());
+                television1.setAvailableSize(updatedTelevision.getAvailableSize());
+                television1.setAmbiLight(updatedTelevision.getAmbiLight());
+                television1.setBluetooth(updatedTelevision.getBluetooth());
+                television1.setBrand(updatedTelevision.getBrand());
+                television1.setHdr(updatedTelevision.getHdr());
+                television1.setName(updatedTelevision.getName());
+                television1.setOriginalStock(updatedTelevision.getOriginalStock());
+                television1.setPrice(updatedTelevision.getPrice());
+                television1.setRefreshRate(updatedTelevision.getRefreshRate());
+                television1.setScreenQuality(updatedTelevision.getScreenQuality());
+                television1.setScreenType(updatedTelevision.getScreenType());
+                television1.setSmartTv(updatedTelevision.getSmartTv());
+                television1.setSold(updatedTelevision.getSold());
+                television1.setType(updatedTelevision.getType());
+                television1.setVoiceControl(updatedTelevision.getVoiceControl());
+                television1.setWifi(updatedTelevision.getWifi());
+                Television returnTelevision = televisionRepository.save(television1);
+                return ResponseEntity.ok().body(returnTelevision);
+            }
+        }
+            // Bij de PutMapping was ik niet helemaal zelf uitgekomen, ik heb ik de uitwerkingen gekeken. Dit werkt nu, maar ik heb nog niet het idee dat ik alle onderdelen echt doorgrond.
```

Comment:
Wat ik zie is dat, om en television te updaten, je een nieuw television-object meestuurt met de PUT call. Dan kijkt ie of de television die je wilt updaten wel bestaat, zo ja dan haalt ie alle properties uit het meegestuurde object en vervangt daarmee de properties van het originele, gevonden television-object. Dat wordt dan in de database opgeslagen en uiteindelijk ook in de respons van de call getoond.